trae_agent/utils/config.py

```diff
@@ -1,210 +1,342 @@
-    Args:
-        config_file: the relative path of your config file, default setting would be trae_config.json
-        provider: default provider is openai, currently only support openai and claude
-        model: the model that you want to use
-        model_base_url: the base url of the model
-        api_key: your api key
-        maximum_step: maximum number of step of the agent. Default setting is 20
-
-    Return:
-        Config Object
+    Base class for agent configurations.
     """
 
-    config: Config = Config(config_file)
+    max_steps: int
+    model: ModelConfig
+    tools: list[str]
 
-    resolved_provider = resolve_config_value(provider, config.default_provider) or "openai"
-    config.default_provider = str(resolved_provider)
 
-    # Resolve configuration values with CLI overrides
-    resolved_model = resolve_config_value(
-        model, config.model_providers[str(resolved_provider)].model
+@dataclass
+class TraeAgentConfig(AgentConfig):
+    """
+    Trae agent configuration.
+    """
+
+    enable_lakeview: bool = True
+    tools: list[str] = field(
+        default_factory=lambda: [
+            "bash",
+            "str_replace_based_edit_tool",
+            "sequentialthinking",
+            "task_done",
+        ]
     )
 
-    model_parameters = config.model_providers[str(resolved_provider)]
-    if resolved_model is not None:
-        model_parameters.model = str(resolved_model)
+    def resolve_config_values(
+        self,
+        *,
+        max_steps: int | None = None,
+    ):
+        resolved_value = resolve_config_value(cli_value=max_steps, config_value=self.max_steps)
+        if resolved_value:
+            self.max_steps = int(resolved_value)
 
-    # Map providers to their environment variable names
-    env_var_api_key = str(resolved_provider).upper() + "_API_KEY"
-    env_var_api_base_url = str(resolved_provider).upper() + "_BASE_URL"
 
-    resolved_api_key = resolve_config_value(
-        api_key,
-        config.model_providers[str(resolved_provider)].api_key,
-        env_var_api_key,
-    )
+@dataclass
+class LakeviewConfig:
+    """
+    Lakeview configuration.
+    """
 
-    resolved_api_base_url = resolve_config_value(
-        model_base_url,
-        config.model_providers[str(resolved_provider)].base_url,
-        env_var_api_base_url,
-    )
+    model: ModelConfig
 
-    if resolved_api_key is not None:
-        # If None shall we stop the program ?
-        model_parameters.api_key = str(resolved_api_key)
 
-    if resolved_api_base_url is not None:
-        model_parameters.base_url = str(resolved_api_base_url)
+@dataclass
+class Config:
+    """
+    Configuration class for agents, models and model providers.
+    """
+
+    lakeview: LakeviewConfig | None = None
+    model_providers: dict[str, ModelProvider] | None = None
+    models: dict[str, ModelConfig] | None = None
+
+    trae_agent: TraeAgentConfig | None = None
+
+    @classmethod
+    def create(
+        cls,
```

Comment:
Just some idea: 
I think for the create function parsing models, Lakeview and agents can actually split to three private functions. 
```suggestion
        def _config_get_model_providers(yaml_config:Any)->Any:
           model_providers = yaml.get("model_providers", None)
           # try except here 
           return model_providers
```